src/System.Data.OleDb/src/ColumnBinding.cs

```diff
@@ -0,0 +1,1500 @@
+            Debug.Assert(DBStatus.S_OK == StatusValue(), "Value_DATE");
+            return DateTime.FromOADate(RowBinding.ReadDouble(ValueOffset));
+        }
+        private void Value_DATE(DateTime value) {
+            Debug.Assert(NativeDBType.DATE == DbType, "Value_DATE");
+            LengthValue(0);
+            StatusValue(DBStatus.S_OK);
+            RowBinding.WriteDouble(ValueOffset, value.ToOADate());
+        }
+
+        private DateTime Value_DBDATE() {
+            Debug.Assert(NativeDBType.DBDATE == DbType, "Value_DBDATE");
+            Debug.Assert(DBStatus.S_OK == StatusValue(), "Value_DBDATE");
+            return RowBinding.ReadDate(ValueOffset);
+        }
+        private void Value_DBDATE(DateTime value) {
+            Debug.Assert(NativeDBType.DBDATE == DbType, "Value_DATE");
+            LengthValue(0);
+            StatusValue(DBStatus.S_OK);
+            RowBinding.WriteDate(ValueOffset, value);
+        }
+
+        private TimeSpan Value_DBTIME() {
+            Debug.Assert(NativeDBType.DBTIME == DbType, "Value_DBTIME");
+            Debug.Assert(DBStatus.S_OK == StatusValue(), "Value_DBTIME");
+            return RowBinding.ReadTime(ValueOffset);
+        }
+        private void Value_DBTIME(TimeSpan value) {
+            Debug.Assert(NativeDBType.DBTIME == DbType, "Value_DBTIME");
+            LengthValue(0);
+            StatusValue(DBStatus.S_OK);
+            RowBinding.WriteTime(ValueOffset, value);
+        }
+
+        private DateTime Value_DBTIMESTAMP() {
+            Debug.Assert(NativeDBType.DBTIMESTAMP == DbType, "Value_DBTIMESTAMP");
+            Debug.Assert(DBStatus.S_OK == StatusValue(), "Value_DBTIMESTAMP");
+            return RowBinding.ReadDateTime(ValueOffset);
+        }
+        private void Value_DBTIMESTAMP(DateTime value) {
+            Debug.Assert(NativeDBType.DBTIMESTAMP == DbType, "Value_DBTIMESTAMP");
+            LengthValue(0);
+            StatusValue(DBStatus.S_OK);
+            RowBinding.WriteDateTime(ValueOffset, value);
+        }
+
+        private Decimal Value_DECIMAL() {
+            Debug.Assert(NativeDBType.DECIMAL == DbType, "Value_DECIMAL");
+            Debug.Assert(DBStatus.S_OK == StatusValue(), "Value_DECIMAL");
+            int[] buffer = new int[4];
+            RowBinding.ReadInt32Array(ValueOffset, buffer, 0, 4);
+            return new Decimal(
+                buffer[2],  // low
+                buffer[3],  // mid
+                buffer[1],  // high
+                (0 != (buffer[0] & unchecked((int)0x80000000))), // sign
+                unchecked((byte)((buffer[0] & unchecked((int)0x00FF0000)) >> 16))); // scale, MDAC 95080
+        }
+        private void Value_DECIMAL(Decimal value) {
+            Debug.Assert(NativeDBType.DECIMAL == DbType, "Value_DECIMAL");
+
+            /* pending breaking change approval
+            if (_precision < ((System.Data.SqlTypes.SqlDecimal) value).Precision) { // WebData 87236
```

Comment:
WebData nnnn are also bug numbers like MDAC

Anyway I guess this comment can go